### DISCUSSION TITLE
Issue to install 4.9 platform vSphere

### DISCUSSION THREAD

**Discussion by EmanueleMiz** [okd-project/okd#1047] [General]
**Describe the bug**
Log bundle :
Warning: Permanently added '10.29.201.30' (ED25519) to the list of known hosts.
Gathering master systemd summary ...
Gathering master failed systemd unit status ...
Gathering master journals ...
Gathering master networking ...
Gathering master containers ...
time="2021-12-20T14:21:10Z" level=warning msg="Failed to decode the keys [\"storage.options.override_kernel_check\"] from \"/etc/containers/storage.conf\"."
Waiting for logs ...

master file /etc/containers/storage.conf :

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

**Version**
4.9.0-0.okd-2021-12-12-025847

**How reproducible**

**Log bundle**
[log-bundle-20211220152100.tar.gz](https://github.com/openshift/okd/files/7745975/log-bundle-20211220152100.tar.gz)

**Comment by vrutkovs** [comment]
```
Dec 20 14:07:02 bootstrap bootkube.sh[9445]:         Pod Status:openshift-kube-apiserver/kube-apiserver        DoesNotExist
Dec 20 14:07:02 bootstrap bootkube.sh[9445]:         Pod Status:openshift-kube-scheduler/openshift-kube-scheduler        DoesNotExist
Dec 20 14:07:02 bootstrap bootkube.sh[9445]:         Pod Status:openshift-kube-controller-manager/kube-controller-manager        DoesNotExist
Dec 20 14:07:02 bootstrap bootkube.sh[9445]:         Pod Status:openshift-cluster-version/cluster-version-operator        Pending
```
There are no schedulable nodes, as they use invalid DNS name:
```
Dec 20 14:21:08 Guezzt-master-1 hyperkube[5047]: E1220 14:21:08.077013    5047 kubelet_node_status.go:95] "Unable to register node with API server" err="Node \"Guezzt-master-1\" is invalid: metadata.name: Invalid value: \"Guezzt-master-1\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')" node="Guezzt-master-1"
```

**Comment by mbudnick** [comment]
[log-bundle-20220103141519.tar.gz](https://github.com/openshift/okd/files/7802146/log-bundle-20220103141519.tar.gz)

I have the same error, but I have only lowercase names. And I don't know that I set override_kernel_check.

**Reply by vrutkovs** [reply]
Different error:
```
Jan 03 15:07:37 localhost bootkube.sh[12263]:         Pod Status:openshift-kube-apiserver/kube-apiserver        RunningNotReady
```
bootstrap should not have `localhost` hostname either

**Reply by mbudnick** [reply]
This is a installation on vSphere in a restricted network with installer-provisioned infrastructure (IPI), so I can not set the hostname for bootstrap.
I found
https://github.com/openshift/okd/issues/407
but this is solved since one year.

**Reply by vrutkovs** [reply]
>I can not set the hostname for bootstrap.

DHCP server should have assigned the hostname to it

**Reply by mbudnick** [reply]
installer clones bootstrap machine from clusterOSImage in vSphere and so the machine has a random MAC.
I find no parameter in install-config.yaml or created manifests to determine the MAC, so I can not tell DHCP server a MAC to assigne a hostname.

**Reply by mbudnick** [reply]
Wtih an own dhcp-sever which send "option host-name" the installation is successfull.
There should be a hint in "Networking requirements" for this in
https://docs.okd.io/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned.html
